### PR TITLE
refactoring checkbox component

### PR DESCRIPTION
**Pull request by OsamaAbdellateef**

### CODE REVIEW

File: packages/components/src/Checkbox/CheckboxGroup.test.tsx [hunk 1]
```diff
@@ -122,24 +93,22 @@ test('passes down "read-only" to checkboxes', () => {
...
+      <Checkbox value="three" data-testid="three">
+        three
+      </Checkbox>
+    </CheckboxGroup>
   );
 
-  expect(screen.getByText('one')).toHaveClass(`data-[error]:bg-red-500`);
-  expect(screen.getByText('two')).toHaveClass(`data-[error]:bg-red-500`);
-  expect(screen.getByText('three')).toHaveClass(`data-[error]:bg-red-500`);
+  // expect(screen.getByText('one')).toHaveClass(`data-[error]:bg-red-500`);
```
Comment (sebald) [inline packages/components/src/Checkbox/CheckboxGroup.test.tsx:109]: ehem :D

**Review by sebald — CHANGES_REQUESTED** [review]
Just some minor things that need to be adjusted 💪

File: packages/components/src/Checkbox/Checkbox.test.tsx [hunk 2]
```diff
@@ -1,14 +1,15 @@
...
+import { Theme } from '@marigold/system';
 import { Checkbox } from './Checkbox';
-import { checkbox } from '../../../../themes/theme-core/src/components/Checkbox.styles';
+import { Checkbox as checkboxStyle } from '../../../../themes/theme-core/src/components/Checkbox.styles';
+import { setup } from '../test.utils';
 
 const theme: Theme = {
   name: 'test',
   components: {
-    Checkbox: checkbox,
```
[the remark this answers is not in the document]
Comment (OsamaAbdellateef) [reply packages/components/src/Checkbox/Checkbox.test.tsx:11]: carbon (4)
 oh i got ur point, 
 is it ok to write them like this !

File: packages/components/src/Checkbox/Checkbox.test.tsx [hunk 3]
```diff
@@ -59,16 +62,12 @@ test('supports read only state', () => {
...
-    <ThemeProvider theme={theme}>
-      <Checkbox data-testid="checkbox">With Label</Checkbox>
-    </ThemeProvider>
-  );
+  render(<Checkbox data-testid="checkbox">With Label</Checkbox>);
 
   const label = screen.getByText('With Label');
 
   expect(label.parentElement).toHaveClass(
-    'flex item-center gap-[1ch] relative',
```
[the remark this answers is not in the document]
Comment (OsamaAbdellateef) [reply packages/components/src/Checkbox/Checkbox.test.tsx:71]: this is  too much easier, thank you  :)

File: packages/components/src/Checkbox/CheckboxGroup.test.tsx (hunk 1 above)
Comment (OsamaAbdellateef) [reply packages/components/src/Checkbox/CheckboxGroup.test.tsx:127]: Actually there's no red color for errors in `theme-core` 
Is that right  !

Comment (sarahgm) [reply packages/components/src/Checkbox/CheckboxGroup.test.tsx:127]: there is one in colors.ts:
```
error: {
    text: '#f33',
    bg: '#f33',
  },
```

Comment (OsamaAbdellateef) [reply packages/components/src/Checkbox/CheckboxGroup.test.tsx:127]: But why should the `checkbox` have theses classes 
the red color should be for helper text not the `checkbox` itself

Comment (sarahgm) [reply packages/components/src/Checkbox/CheckboxGroup.test.tsx:127]: so just remove the comments

**Comment by sarahgm** [comment]
@OsamaAbdellateef  conflicts

**Comment by OsamaAbdellateef** [comment]
> @OsamaAbdellateef conflicts

Resolved :heavy_check_mark:

**Review by sarahgm — APPROVED** [review]

**Pull request merged**